### PR TITLE
feat(cli/standalone): Support runtime flags for deno compile

### LINKED ISSUE #8656: `deno compile` does not provide the ability to specify permissions or v8 flags
Currently all code run in `deno compile` effectively uses `--allow-all` permissions, and the default `--v8-flags`. This issue exists to investigate if there is demand for specifying permissions or v8 flags for `deno compile`.

### PR DESCRIPTION
**Pull request by nayeemrmn**
Supports permissions etc for executables created with `deno compile`, by appending serialized `Flags` after the source bundle.

Closes #8656.

### CODE REVIEW

File: cli/flags.rs
Comment (bartlomieju) [inline cli/flags.rs:32]: This is surprising; it means compiled binaries can't have runtime args. What's the use case for having pre compiled CLI args?

Comment (nayeemrmn) [reply cli/flags.rs:32]: > This is surprising; it means compiled binaries can't have runtime args. What's the use case for having pre compiled CLI args?

No, the runtime args are _appended_ to the ones passed at compile-time, like with `deno install`.

Comment (bartlomieju) [reply cli/flags.rs:32]: Okay, good to know, can you update docs page as well?

**Comment by bartlomieju** [comment]
@nayeemrmn anything holding back this PR?

**Comment by nayeemrmn** [comment]
Done now, I still had to do ca file bundling.

File: cli/main.rs
Comment (bartlomieju) [inline cli/main.rs:317]: Maybe `inspect` and `inspect_brk` shouldn't be available to `compile` subcommand at all?

File: cli/standalone.rs
Comment (bartlomieju) [inline cli/standalone.rs:66]: It'd be useful for debugging to add `.context()` before `?` with description what went wrong

File: cli/standalone.rs
Comment (bartlomieju) [inline cli/standalone.rs:81]: It'd be useful for debugging to add .context() before ? with description what went wrong

File: cli/standalone.rs
Comment (bartlomieju) [inline cli/standalone.rs:150]: Please add a note that it should be kept in sync with the identical call in `main.rs`

File: cli/standalone.rs
Comment (bartlomieju) [inline cli/standalone.rs:307]: Adding `watch` makes no sense in compiled binaries

File: cli/standalone.rs
Comment (bartlomieju) [inline cli/standalone.rs:272]: Add `IMPORTANT: ` at the begging please :D

**Review by bartlomieju — COMMENTED** [review]
@nayeemrmn awesome work! I have only a few minor comments but in general LGTM. 

I'll wait for @lucacasonato's review before landing

File: cli/file_fetcher.rs
Comment (lucacasonato) [inline cli/file_fetcher.rs:286]: Can we rename this to ca_data to match the impl in op_crates/fetch?

File: cli/flags.rs
Comment (lucacasonato) [inline cli/flags.rs:110]: Why keep this if we have ca_pem? I would prefer to have just one of the two.

File: cli/standalone.rs
Comment (lucacasonato) [inline cli/standalone.rs:43]: I don't think this is needed.

File: cli/main.rs
Comment (nayeemrmn) [reply cli/main.rs:317]: > Maybe `inspect` and `inspect_brk` shouldn't be available to `compile` subcommand at all?

I included them because it seems simpler to group them into the wider set of "flags that affect runtime behaviour", and match `deno install` in doing so. On the other hand, is there potential to remove the inspector from the light runtime binary?

Comment (caspervonb) [reply cli/main.rs:317]: > On the other hand, is there potential to remove the inspector from the light runtime binary?

We can feature gate it on our side; it will break features like the REPL but those seem irrelevant for compiled bundles.

**Review by lucacasonato — COMMENTED** [review]
Just a few comments.

I don't like the mix of compile args and runtime args. Same goes for `deno install` though, so it is fine (not a blocker).

File: cli/main.rs
Comment (bartlomieju) [reply cli/main.rs:317]: > On the other hand, is there potential to remove the inspector from the light runtime binary?

Not in the near future, no.

> We can feature gate it on our side; it will break features like the REPL but those seem irrelevant for compiled bundles.

REPL is not part of `runtime/` directory, it's `cli/` only so that should work fine

File: cli/standalone.rs
Comment (nayeemrmn) [reply cli/standalone.rs:43]: I thought the redundant values would make the extraction more readable, but it's not that different. Switching to only use the pointers.

File: cli/flags.rs
Comment (nayeemrmn) [reply cli/flags.rs:110]: We'd have to read the file in flag parsing, and it wouldn't correspond well to the actual CLI flags. ~`ca_pem`~ `ca_data` is the one that doesn't belong here, but it's a particularly convenient hack.

File: cli/main.rs
Comment (nayeemrmn) [reply cli/main.rs:317]: Doh, apparently I already went out of my way to exclude the inspector flags in `flags.rs` 🤦‍♂️ I'll just remove this meaningless handling.

File: cli/main.rs
Comment (nayeemrmn) [inline cli/main.rs:140]: Fix before landing.

I've moved things around a bit so this is the only one -- @bartlomieju any ideas on how to return a `Result` in this section?

**Review by nayeemrmn — COMMENTED** [review]
I've removed the `Flags::ca_data` hack, since nothing that directly reads program state's flags should be used in a compiled runtime.

File: cli/main.rs
Comment (bartlomieju) [reply cli/main.rs:140]: @nayeemrmn you'll have to change type of `CreateWebWorkerCb`, I'm fine with that

File: runtime/web_worker.rs
Comment (lucacasonato) [inline runtime/web_worker.rs:142]: Why is this needed in the `WebWorkerOptions`? Can't we read the file during initialization?

File: runtime/worker.rs
Comment (lucacasonato) [inline runtime/worker.rs:50]: Ditto

File: cli/main.rs
Comment (nayeemrmn) [reply cli/main.rs]: Done, it's read and stored in `ProgramState`.

**Review by lucacasonato — APPROVED** [review]
LGMT. Very clean implementation now :+1:

**Review by bartlomieju — APPROVED** [review]
LGTM

**Pull request merged**
